File: MacOS/Type4Me/ASR/Providers/IflytekASRConfig.swift
import Foundation

struct IflytekASRConfig: ASRProviderConfig, Sendable {

    static let provider = ASRProvider.iflytek
    static var displayName: String { L("讯飞", "iFLYTEK") }

    static var credentialFields: [CredentialField] {[
        CredentialField(key: "appId", label: "App ID", placeholder: L("应用 ID", "App ID"), isSecure: false, isOptional: false, defaultValue: ""),
        CredentialField(key: "apiKey", label: "API Key", placeholder: L("密钥", "Secret"), isSecure: true, isOptional: false, defaultValue: ""),
        CredentialField(key: "apiSecret", label: "API Secret", placeholder: L("密钥", "Secret"), isSecure: true, isOptional: false, defaultValue: ""),
    ]}

    let appId: String
    let apiKey: String
    let apiSecret: String

    init?(credentials: [String: String]) {
        guard let aid = credentials["appId"], !aid.isEmpty,
              let key = credentials["apiKey"], !key.isEmpty,
              let secret = credentials["apiSecret"], !secret.isEmpty
        else { return nil }
        self.appId = aid
        self.apiKey = key
        self.apiSecret = secret
    }

    func toCredentials() -> [String: String] {
        ["appId": appId, "apiKey": apiKey, "apiSecret": apiSecret]
    }

    var isValid: Bool { !appId.isEmpty && !apiKey.isEmpty && !apiSecret.isEmpty }
}

File: Windows/Type4Me/ASR/Providers/IflytekASRConfig.cs
using Type4Me.Localization;

namespace Type4Me.ASR.Providers;

public sealed class IflytekASRConfig
{
    public string AppId { get; }
    public string ApiKey { get; }
    public string ApiSecret { get; }

    public static ASRProvider Provider => ASRProvider.Iflytek;
    public static string DisplayName => Loc.L("讯飞", "iFLYTEK");

    public static CredentialField[] CredentialFields =>
    [
        new() { Key = "appId", Label = "App ID", Placeholder = Loc.L("应用 ID", "App ID") },
        new() { Key = "apiKey", Label = "API Key", Placeholder = Loc.L("密钥", "Secret"), IsSecure = true },
        new() { Key = "apiSecret", Label = "API Secret", Placeholder = Loc.L("密钥", "Secret"), IsSecure = true },
    ];

    private IflytekASRConfig(string appId, string apiKey, string apiSecret)
    {
        AppId = appId; ApiKey = apiKey; ApiSecret = apiSecret;
    }

    public static IflytekASRConfig? TryCreate(Dictionary<string, string> credentials)
    {
        if (!credentials.TryGetValue("appId", out var aid) || string.IsNullOrEmpty(aid)) return null;
        if (!credentials.TryGetValue("apiKey", out var key) || string.IsNullOrEmpty(key)) return null;
        if (!credentials.TryGetValue("apiSecret", out var secret) || string.IsNullOrEmpty(secret)) return null;
        return new IflytekASRConfig(aid, key, secret);
    }

    public Dictionary<string, string> ToCredentials() => new()
    {
        ["appId"] = AppId, ["apiKey"] = ApiKey, ["apiSecret"] = ApiSecret,
    };

    public bool IsValid => !string.IsNullOrEmpty(AppId) && !string.IsNullOrEmpty(ApiKey) && !string.IsNullOrEmpty(ApiSecret);
}
